kyo-prelude/shared/src/main/scala/kyo/kernel/Flat.scala

```diff
@@ -79,10 +79,10 @@ private object FlatMacro:
                     base =:= TypeRepr.of[Duration]
                 case _ => false
 
-        def hasTag(t: TypeRepr): Boolean =
+        def canDerive(t: TypeRepr): Boolean =
             t.asType match
                 case '[t] =>
-                    Expr.summon[Tag.Full[t]].isDefined
+                    Expr.summon[Tag.Full[t]].isDefined || Expr.summon[Flat[t]].isDefined
```

Comment:
Doesn't it make sense to choose the evidence first?

Reply:
good point! It should be more lightweight than invoking the tag macro first